packages/mds-stream/index.ts

```diff
@@ -199,7 +201,10 @@ async function writeStreamBatch(stream: Stream, field: string, values: unknown[]
 // put basics of vehicle in the cache
 async function writeDevice(device: Device) {
   if (env.NATS) {
-    return writeNatsEvent('device', JSON.stringify(device))
+    await writeNatsEvent('device', JSON.stringify(device))
+  }
+  if (env.KAFKA_HOST) {
+    await KafkaStream.writeDevice(device)
   }
   return writeStream(DEVICE_INDEX_STREAM, 'data', device)
```

Comment:
Seems like an opportunity for a `Promise.all([...])` in here.

Reply:
For sure. I can't find a very clean way of writing it without refactoring much of the nats/redis methods. Once they all expose methods with the same arguments, it will be easier.